virtcontainers/sandbox.go

```diff
@@ -2150,32 +2150,43 @@ func (s *Sandbox) cpuResources() *specs.LinuxCPU {
 
 // setupSandboxCgroup creates and joins sandbox cgroups for the sandbox config
 func (s *Sandbox) setupSandboxCgroup() error {
+	var err error
 	spec := s.GetPatchedOCISpec()
-
 	if spec == nil {
 		return errorMissingOCISpec
 	}
 
-	if spec.Linux == nil {
-		// Cgroup path is optional, though expected. If not defined, skip the setup
-		s.Logger().WithField("sandboxid", s.id).Warning("no cgroup path provided for pod sandbox, not creating sandbox cgroup")
-		return nil
+	s.state.CgroupPath, err = validCgroupPath(spec.Linux.CgroupsPath, s.config.SystemdCgroup)
+	if err != nil {
+		return fmt.Errorf("Invalid cgroup path: %v", err)
 	}
-	validContainerCgroup := utils.ValidCgroupPath(spec.Linux.CgroupsPath)
 
-	// Create a Kata sandbox cgroup with the cgroup of the sandbox container as the parent
-	s.state.CgroupPath = filepath.Join(filepath.Dir(validContainerCgroup), cgroupKataPrefix+"_"+s.id)
```

Comment:
Can you add an explanation in the commit why this is no longer required. 
In general, I would like to see a design doc for this PR showing diagramatically how we create cgroups in v1 vs v2.

Reply:
this is still required, it was just moved, `validCgroupPath` joins the paths if needed

Reply:
> how we create cgroups in v1 vs v2.

we rely on libcontainer, it detects the cgroup version and create a manager for it